src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +364,177 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(self, task_execution: _TaskExecution):
+        result = TaskResult(
+            outcome_code=TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE,
+            failure_reason=TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR,
+            stderr="",
+            stdout="",
+        )
+
+        def progress_reporter(progress: Progress):
+            with task_execution.updated:
+                task_execution.progress = ProgressUpdate(
+                    current=progress.current, total=progress.total
+                )
+                task_execution.updated.notify_all()
+            # sleep(0) here momentarily releases the GIL, giving
+            # other threads a chance to run.
+            time.sleep(0)
+
+            # TODO: Consider implementing task cancellation here.
+
+        try:
+            # Run the task handler
+            result = RunTaskHandler(
+                task=task_execution.task,
+                invocation_state=ProxiedInvocationState(
+                    task_execution.task.task_id, self._invocation_state_proxy_server
+                ),
+                function_wrapper=self._function_wrapper,
+                function_stdout=self._function_stdout,
+                function_stderr=self._function_stderr,
+                graph_metadata=self._graph_metadata,
+                progress_reporter=progress_reporter,
+                logger=self._logger,
+            ).run()
+
+            if result.outcome_code == TaskOutcomeCode.TASK_OUTCOME_CODE_UNKNOWN:
+                result.outcome_code = TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE
```

Comment:
Please add an error log line explaining that there's an unexpected service state/bug here.

Reply:
I just removed it - we'll count on the invariants.